source/docs/software/hardware-apis/sensors/encoders-software.rst

```diff
@@ -226,15 +236,223 @@ To reset an encoder to a distance reading of zero, call the :code:`reset()` meth
         // Resets the encoder to read a distance of zero
         encoder.Reset();
 
+Duty Cycle Encoders - The :code:`DutyCycleEncoder` class
+--------------------------------------------------------
+
+WPILib provides support for duty cycle (also marketed as PWM) encoders through the :code:`DutyCycleEncoder` class (`Java <https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/DutyCycleEncoder.html>`__, `C++ <https://github.wpilib.org/allwpilib/docs/release/cpp/classfrc_1_1_duty_cycle_encoder.html>`__).  This class provides a simple API for configuring and reading data from duty cycle encoders.
+
+The roboRIO's FPGA handles duty cycle encoders automatically.
+
+Examples of duty cycle encoders:
+
+- `AndyMark Mag Encoder <https://www.andymark.com/products/am-mag-encoder>`__
+- `CTRE Mag Encoder <https://store.ctr-electronics.com/srx-mag-encoder/>`__
+- `REV Through Bore Encoder <https://www.revrobotics.com/rev-11-1271/>`__
+- `US Digital MA3 <https://www.usdigital.com/products/encoders/absolute/shaft/ma3/>`__
+
+Initializing a duty cycle encoder
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A duty cycle encoder can be instantiated as follows:
+
+.. tabs::
+
+    .. code-tab:: java
+
+        // Initializes a duty cycle encoder on DIO pins 0
+        DutyCycleEncoder encoder = new DutyCycleEncoder(0);
+
+    .. code-tab:: c++
+
+        // Initializes a duty cycle encoder on DIO pins 0
+        frc::DutyCycleEncoder encoder{0};
+
+Configuring duty cycle encoder parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note:: The :code:`DutyCycleEncoder` class does not make any assumptions about units of distance; it will return values in whatever units were used to calculate the distance-per-rotation value.  Users thus have complete control over the distance units used.
+
+The :code:`DutyCycleEncoder` class offers a number of configuration methods:
+
+.. tabs::
+
+    .. code-tab:: java
+
+        // Configures the encoder to return a distance of 4 for every rotation
+        encoder.setDistancePerRotation(4.);
+
+    .. code-tab:: c++
+
+        // Configures the encoder to return a distance of 4 for every rotation
+        encoder.SetDistancePerRotation(4.);
```

Comment:
This method exists?

Reply:
https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/AnalogEncoder.html#setDistancePerRotation(double)